File: Habitica/src/main/java/com/habitrpg/android/habitica/data/ContentRepository.java
package com.habitrpg.android.habitica.data;

import com.habitrpg.android.habitica.models.ContentResult;
import com.habitrpg.android.habitica.models.WorldState;

import io.reactivex.Flowable;

public interface ContentRepository extends BaseRepository {

    Flowable<ContentResult> retrieveContent();
    Flowable<ContentResult> retrieveContent(boolean forced);

    Flowable<WorldState> retrieveWorldState();
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/data/ContentRepository.kt
package com.habitrpg.android.habitica.data

import com.habitrpg.android.habitica.models.ContentResult
import com.habitrpg.android.habitica.models.WorldState
import kotlinx.coroutines.flow.Flow

interface ContentRepository: BaseRepository {
    suspend fun retrieveContent(forced: Boolean = false): ContentResult?

    suspend fun retrieveWorldState(forced: Boolean = false): WorldState?
    fun getWorldState(): Flow<WorldState>
}
